src/views/PluginList.tsx

```diff
@@ -57,13 +66,21 @@ interface IBaseProps {
   gameSupported: (gameMode: string) => boolean;
   minRevision: (gameMode: string) => number;
   supportsESL: (gameMode: string) => boolean;
+  revisionText: (gameMode: string) => string;
   getPluginFlags(
     t: TranslationFunction,
     plugin: IPluginCombined,
     gameSupported: boolean,
     supportsESL: boolean,
     minRevision: number
   ): string[];
+  isMaster: (filePath: string, flag: boolean, gameMode: string) => boolean;
+  isLight: (filePath: string, flag: boolean, gameMode: string) => boolean;
+  openLOOTSite: () => Promise<any>;
+  getNewESPFile: (filePath: string) => ESPFile;
```

Comment:
This doesn't seem type-save, this ESPFile is a separate type from what the function in index.ts returns.
At the very least:
a) turn the ESPFile type in this file into an interface IESPFile
b) export that interface
c) in index.ts, actually return a regular javascript object that implements that interface. Copy the values as necessary.

Further, I would probably drop the setLightFlag member function from this type and instead pass in an additional prop to the component, something like `setPluginLight: (filePath: string, enabled: boolean) => Promise<void>;`

That way you have more abstraction between the component interface and the external library interface. You would have to parse the ESPFile again in setPluginLight but I don't think that's a performance concern we have to worry about.

Also, why is it named "getNewESPFile"? It parses an esp file, it doesn't generate a new one.